meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockClient.java

```diff
@@ -64,8 +64,12 @@ public LockClient(MetaClientInterface<LockInfo> client) {
       throw new IllegalArgumentException("MetaClient cannot be null.");
     }
     _metaClient = client;
-    LOG.info("Connecting to existing MetaClient for LockClient");
-    _metaClient.connect();
+    try {
+      LOG.info("Connecting to existing MetaClient for LockClient");
+      _metaClient.connect();
+    } catch (IllegalStateException e) {
+      // Ignore as it either has already been connected or already been closed.
```

Comment:
One question. If the client is already closed, what exception do metaclient throw? If it is also IllegalStateException, we should considering differentiate calling `connect` on already connected client or on already closed client.

Reply:
Will create a todo for the next PR to address this!